src/Objects/_PVector.js

```diff
@@ -0,0 +1,283 @@
+             len === 2? _PVector.dot(v, y) :
+             len > 2? this.x*v + this.y*y + this.z*z :
+             argsErr('dot', len, 1);
+    },
+    cross: function(v1, v2, target) {
+      return target? _PVector.cross(v1, v2, target)
+                   : _PVector.cross(this, v1, v2);
+    },
+    lerp: function() {
+      var x, y, z, amt;
+      const len = arguments.length;
+      if ((len | 1) === 1) { argsErr('lerp', len, 2); }
+      if (len === 2) { // given vector and amt
+        const v = arguments[0];
+        x = v.x;
+        y = v.y;
+        z = v.z;
+        amt = arguments[1];
+      } else if (len === 3) { // given vector 1, vector 2 and amt
+        return _PVector.lerp(arguments[0], arguments[1], arguments[2]);
+      } else { // given x, y, z and amt
+        x = arguments[0];
+        y = arguments[1];
+        z = arguments[2];
+        amt = arguments[3];
+      }
+      this.x = lerp(this.x, x, amt);
+      this.y = lerp(this.y, y, amt);
+      this.z = lerp(this.z, z, amt);
+      return this;
+    },
+    add: function(v, y, z) {
+      if (y instanceof _PVector) { return _PVector.add(v, y, z); }
+      const len = arguments.length;
+      if (len === 1) {
+        //_PVector.add(this, v, this);
+        this.x += v.x;
+        this.y += v.y;
+        this.z += v.z;
+      } else if (len > 1) {
+        this.x += v;
+        this.y += y;
+        if (len > 2) { this.z += z };
+      } else { argsErr('add', len, 1); }
+      return this;
+    },
+    sub: function(v, y, z) {
+      if (y instanceof _PVector) { return _PVector.sub(v, y, z); }
+      const len = arguments.length;
+      if (len === 1) {
+        this.x -= v.x;
+        this.y -= v.y;
+        this.z -= v.z;
+      } else if (len > 1) {
+        this.x -= v;
+        this.y -= y;
+        if (len > 2) { this.z -= z; }
+      } else { argsErr('sub', len, 1); }
+      return this;
+    },
+    mult: function(v, n, target) {
+      const len = arguments.length;
+      if (len === 1) {
+        if (typeof v === 'number') {
+          this.x *= v;
+          this.y *= v;
+          this.z *= v;
+        } else {
+          this.x *= v.x;
+          this.y *= v.y;
+          this.z *= v.z;
+        }
+        return this;
+      } else if (len > 1) { return _PVector.mult(v, n, target); }
+      argsErr('mult', len, 1);
+    },
+    div: function(v, n, target) {
+      const len = arguments.length;
+      if (len === 1) {
+        if (typeof v === 'number') {
+          this.x /= v;
+          this.y /= v;
+          this.z /= v;
+        } else {
+          this.x /= v.x;
+          this.y /= v.y;
+          this.z /= v.z;
+        }
+        return this;
+      } else if (len > 1) { return _PVector.div(v, n, target); }
+      argsErr('div', len, 1);
+    },
+    toString: function() {
+      return '[' + this.x + ', ' + this.y + ', ' + this.z + ']';
+    },
+    equals: function(o) {
+      return o === this? true : o instanceof _PVector?
```

Comment:
okay, okay, settle down. No need to keep yelling: if you explain that with normal periods, I will happily go over the words you write in rebuttal and take them seriously. In fact, I'll take them far more seriously if you don't constantly yell at me with exclamation points. This isn't a contest, it's a collaboration. If you think I'm wrong, explain to me why you think that instead of yelling it in my face.

You're right, just checking for `add` might leave the check open to false equality, but it's still important to remember what this library is for: accepting valid Processing source code. So: can we think of usecases where this breaks, compared to working Processing code? Yes, we can. The equality test that I used might flag a custom object with an `add` method as a PVector, without being one. That'd be a bug, so my suggestion is flat our wrong.

So just say that instead of going "This is an equals check!!". Neither of us is perfect, so I will try to help you get this code to a point where it fits the Processing.js methodology, and I would like to hear from you when I do something wrong. Not by being yelled at, but by you taking the time to calmly explain why I am wrong. I don't mind being wrong. Every time I'm wrong, I can learn something. But not if you just yell "This doesn't cut it!" and then not explain why you think that.